src/main/java/ch/uzh/ifi/hase/soprafs23/websockets/TimerWebSocketHandler.java

```diff
@@ -0,0 +1,71 @@
+package ch.uzh.ifi.hase.soprafs23.websockets;
+
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import ch.uzh.ifi.hase.soprafs23.service.GameService;
+import org.springframework.web.socket.CloseStatus;
+import org.springframework.web.socket.TextMessage;
+import org.springframework.web.socket.WebSocketSession;
+import org.springframework.web.socket.handler.TextWebSocketHandler;
+
+public class TimerWebSocketHandler extends TextWebSocketHandler {
+
+    // HashMap that stores a list of sessions for each access code
+    private final HashMap<Integer, ArrayList<WebSocketSession>> webSocketSessions = new HashMap<>();
+
+    // The current value of the timer
+    private int timerValue;
+
+    private final GameService gameService;
+
+    public TimerWebSocketHandler(GameService gameService) {
+        this.gameService = gameService;
+    }
+
+    @Override
+    public void afterConnectionEstablished(WebSocketSession session) throws InterruptedException, IOException {
+        int accessCode = getAccessCode(session);
+        // If the access code is not in the HashMap, add it
+        if (!webSocketSessions.containsKey(accessCode)) {
+            webSocketSessions.put(accessCode, new ArrayList<>());
+        }
+        webSocketSessions.get(accessCode).add(session);
+        timerValue = gameService.getGame(accessCode).getSettings().getRoundTime();
+        startTimer(accessCode);
```

Comment:
This timer restarts everytime someone loads/reloads the page, right? I think the timer should only be started by the first person that loads the page.